crates/astria-cli/src/sequencer/sign.rs

```diff
@@ -0,0 +1,99 @@
+use std::{
+    io::Write,
+    path::{
+        Path,
+        PathBuf,
+    },
+};
+
+use astria_core::{
+    protocol::transaction::v1::TransactionBody,
+    Protobuf,
+};
+use clap_stdin::FileOrStdin;
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+
+use crate::utils::signing_key_from_private_key;
+
+#[derive(clap::Args, Debug)]
+pub(super) struct Command {
+    /// The private key of account being sent from
+    #[arg(long, env = "SEQUENCER_PRIVATE_KEY")]
+    // // TODO: https://github.com/astriaorg/astria/issues/594
+    // // Don't use a plain text private, prefer wrapper like from
+    // // the secrecy crate with specialized `Debug` and `Drop` implementations
+    // // that overwrite the key on drop and don't reveal it when printing.
+    private_key: String,
+    /// Target to to write the signed transaction in pbjson format (omit to write to STDOUT).
+    #[arg(long, short)]
+    output: Option<PathBuf>,
+    /// Forces an overwrite of `--output` if a file at that location exists.
+    #[arg(long, short)]
+    force: bool,
+    /// The source to read the pbjson formatted astra.protocol.transaction.v1.Transaction (use `-`
+    /// to pass via STDIN).
+    input: FileOrStdin,
+}
+
+// The goal of the `sign` CLI command is to take in a `TransactionBody` and to sign with a private
+// key to create a `Transaction`. This signed `Transaction` should be printed to the console in
+// pbjson format.
+impl Command {
+    pub(super) fn run(self) -> eyre::Result<()> {
+        let key = signing_key_from_private_key(self.private_key.as_str())?;
+
+        let filename = self.input.filename().to_string();
+        let transaction_body = read_transaction_body(self.input)
+            .wrap_err_with(|| format!("failed to read transaction body from `{filename}`"))?;
+        let transaction = transaction_body.sign(&key);
+
+        serde_json::to_writer(
```

Comment:
it may be useful to have a `--pretty` flag here to use `to_write_pretty` for more human readability

Reply:
This seems to be slightly controversial, so I think we should leave this out for now.

Reply:
To explain my reasoning: I'd rather not have extra functionality that needs to be maintained if the same result can be achieved using standard tooling that's widely available (or can be easily installed).